### ISSUE TITLE
Distributed memory table

### ISSUE THREAD

**Issue by mposdev21** [ClickHouse/ClickHouse#61270]
I want to be able to run aggregations on certain columns based on certain rows which are filtered by user selecting the filter criteria. The filter includes a few columns, time window etc. The expensive part of the query is to filter the relevant rows. Once these rows are obtained the aggregation runs very fast.  I decided to use the Memory table for caching the intermediate results before aggregation.

The main table uses ReplicatedMergeTree engine. For caching the filtered rows, I decided to go with a "Memory" engine where the results from each node would be cached locally. Then run a distributed query for the aggregations based on the cached rows in memory. But from what I have tried, this is not supported.

I am able to create a memory table in the cluster:

**CREATE TABLE IF NOT EXISTS table_local ON CLUSTER '{cluster}' { ...} ENGINE = Memory**

I want to be able to use a Distributed table on top of this:

**CREATE TABLE IF NOT EXISTS table_distributed ON CLUSTER '{cluster}' AS table_local ENGINE = Distributed('{cluster}', default, table_local, rand())**

This command successfully completes. I had to use default as I am not sure what the right value here for Memory table would be.

Then I was able to insert data into this distributed table:

INSERT INTO table_distributed select * from <some_other_table>

Till here the commands did not fail.  Now, if I send a SELECT query to the load balancer, it fails.  If I send this to one specific pod address, it succeeds. It is not clear whether this is a supported use case or not. I could not find an alternative solution as I cannot pre-compute these aggregations till the user specifies the filter criteria.

**Comment by wibed** [comment]
selfish bump. per documentation: https://clickhouse.com/docs/en/engines/table-engines/special/memory
tldr: "data is not replicated across all nodes (by design)"

but still following this for reasons
